stratosphere/pm/source/impl/pm_process_info.hpp

```diff
@@ -0,0 +1,310 @@
+                this->ClearFlag(Flag_ExceptionWaitingAttach);
+            }
+
+            void SetSignalOnDebugEvent() {
+                this->SetFlag(Flag_SignalOnDebugEvent);
+            }
+
+            void SetSuspendedStateChanged() {
+                this->SetFlag(Flag_SuspendedStateChanged);
+            }
+
+            void ClearSuspendedStateChanged() {
+                this->ClearFlag(Flag_SuspendedStateChanged);
+            }
+
+            void SetSuspended() {
+                this->SetFlag(Flag_Suspended);
+            }
+
+            void ClearSuspended() {
+                this->ClearFlag(Flag_Suspended);
+            }
+
+            void SetApplication() {
+                this->SetFlag(Flag_Application);
+            }
+
+            void SetSignalOnStart() {
+                this->SetFlag(Flag_SignalOnStart);
+            }
+
+            void ClearSignalOnStart() {
+                this->ClearFlag(Flag_SignalOnStart);
+            }
+
+            void SetStartedStateChanged() {
+                this->SetFlag(Flag_StartedStateChanged);
+            }
+
+            void ClearStartedStateChanged() {
+                this->ClearFlag(Flag_StartedStateChanged);
+            }
+
+            bool HasStarted() const {
+                return this->state != ProcessState_Created && this->state != ProcessState_CreatedAttached;
+            }
+
+            bool HasExited() const {
+                return this->state == ProcessState_Exited;
+            }
+
+            bool ShouldSignalOnExit() const {
+                return this->HasFlag(Flag_SignalOnExit);
+            }
+
+            bool HasExceptionOccurred() const {
+                return this->HasFlag(Flag_ExceptionOccurred);
+            }
+
+            bool HasExceptionWaitingAttach() const {
+                return this->HasFlag(Flag_ExceptionWaitingAttach);
+            }
+
+            bool ShouldSignalOnDebugEvent() const {
+                return this->HasFlag(Flag_SignalOnDebugEvent);
+            }
+
+            bool ShouldSignalOnStart() const {
+                return this->HasFlag(Flag_SignalOnStart);
+            }
+
+            bool HasSuspendedStateChanged() const {
+                return this->HasFlag(Flag_SuspendedStateChanged);
+            }
+
+            bool IsSuspended() const {
+                return this->HasFlag(Flag_Suspended);
+            }
+
+            bool IsApplication() const {
+                return this->HasFlag(Flag_Application);
+            }
+
+            bool HasStartedStateChanged() const {
+                return this->HasFlag(Flag_StartedStateChanged);
+            }
+
+    };
+
+    Result OnProcessSignaled(std::shared_ptr<ProcessInfo> process_info);
+
+    class ProcessInfoWaiter final : public IWaitable {
+        private:
+            std::shared_ptr<ProcessInfo> process_info;
+        public:
+            ProcessInfoWaiter(std::shared_ptr<ProcessInfo> p) : process_info(p) { /* ... */ }
```

Comment:
Probably doesn't matter, but:

```suggestion
            ProcessInfoWaiter(std::shared_ptr<ProcessInfo> p) : process_info(std::move(p)) { /* ... */ }
```

Reply:
mm, yeah. Probably doesn't matter but you're right.